File: FrontEnd_Angular/src/app/body/api/consultation/Cristallin.ts
export interface Cristallin {
    id: number;
    createdBy: string;
    createdAt: string;
    modifiedBy: string;
    modifiedAt: string;
    deleted: boolean;
    version: number;

    eyeSide: 'OU' | 'OD' | 'OS' | null;
    normal: boolean;
    cataract: boolean;
    cortical: number;
    nuclear: number;
    white: boolean;
    trauma: boolean;
    phacodonesis: boolean;
    luxation: boolean;
    icl: boolean;
    iol: boolean;
    other: boolean;
    otherText?: string | null;
}

File: Beta_1_BackEnd_SpringBoot/src/main/java/com/jorami/eyeapp/model/consultation/Cristallin.java
package com.jorami.eyeapp.model.consultation;

import com.jorami.eyeapp.model.Enum;
import com.jorami.eyeapp.model.IdentifiedModel;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Where;
import org.hibernate.envers.Audited;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Audited
@Where(clause = "deleted = FALSE")
@Entity
@Table(name="cristallin")
public class Cristallin extends IdentifiedModel {
    @Enumerated(EnumType.STRING)
    private Enum.EyeSide eyeSide;
    private boolean normal;
    private boolean cataract;
    private int cortical;
    private int nuclear;
    private boolean white;
    private boolean trauma;
    private boolean phacodonesis;
    private boolean luxation;
    private boolean icl;
    private boolean iol;
    private boolean other;
    private String otherText;
}
